src/main/java/org/broadinstitute/dsm/jobs/ExternalShipperJob.java

```diff
@@ -25,9 +25,7 @@ public void execute(JobExecutionContext context) {
             try {
                 logger.info("Starting the external shipper job");
                 ExternalShipper shipper = (ExternalShipper) Class.forName(DSMServer.getClassName(kitType.getExternalShipper())).newInstance();//GBFRequestUtil
-                ArrayList<KitRequest> kitRequests = shipper.getKitRequestsNotDone(kitType.getInstanceId());
-                shipper.orderStatus(kitRequests);
-                kitRequests.clear();
+                shipper.updateOrderStatusForPendingKitRequests(kitType.getInstanceId());
```

Comment:
Instead of creating a large object here and passing it around, this changes things to streaming.